src/mainClass.js

```diff
@@ -0,0 +1,27 @@
+import { getData, getCategories } from './source';
+
+export class AllItems {
+  constructor(searchCategory, showLinks) {
+    this.searchCategory = searchCategory;
+    this.showLinks = showLinks;
+    this.categories = [];
+  }
+
+  displayCategories() {
+    getCategories().then((res) => {
+      this.categories = res;
+      this.searchCategory(this.categories);
+    });
+  }
+
+  getCategoryItems(category) {
+    getData(category).then((res) => {
+      res.map((el) => {
```

Comment:
Don't use map where is appropriate for or forEach, here you need for looping not map, map has result.
Also don't iterate for ui here, you only call function which is UI related and has own looping, like search category.